dsgrid/project.py

```diff
@@ -110,39 +105,9 @@ def unload_dataset(self, dataset_id):
         dataset = self.get_dataset(dataset_id)
         dataset.delete_views()
 
-    """
-    The code below is subject to change.
-    """
-
-    def _iter_base_dimensions(self):
-        for dimension in self.config.dimensions.base_dimensions:
-            yield dimension
-
     def _iter_datasets(self):
         for dataset in self.config.datasets:
             yield dataset
 
     def list_datasets(self):
         return [x.dataset_id for x in self._iter_datasets()]
-
-    def get_project_dimension(self, dimension_type):
-        for dimension in self._iter_base_dimensions():
-            if dimension.dimension_type == dimension_type:
-                return dimension
-        raise DSGInvalidField(f"{dimension_type} is not stored")
-
-    def get_input_dataset(self, dataset_id):
-        for dataset in self._iter_datasets():
-            if dataset.dataset_id == dataset_id:
-                return dataset
-        raise DSGInvalidField(f"{dataset_id} is not stored")
-
-    def get_geography(self):
-        # TODO: is name right? cls?
-        return self.get_project_dimension(DimensionType.GEOGRAPHY).name
-
-    def get_dimension_mappings(self, dimension_type, mapping_type):
-        if isinstance(mapping_type, str):
-            mapping_type = MappingType(mapping_type)
-        dimension = self.get_project_dimension(dimension_type)
-        return getattr(dimension, mapping_type.value)
```

Comment:
Why are we removing so much code?

Reply:
This is old stuff that we will never use. I didn't want to mess with configuring flake8 to ignore them.

Reply:
But some of these codes we can use in project query. I definitely used a few of them

Reply:
I searched the repo and didn't see any usages. Let's take them one-by-one:
- get_input_dataset: This is actually confusing because it returns a DatasetConfig. There is a get_dataset method that returns a Dataset. I'm improving that one in my query branch. I think that I should go further and either delete or rename list_datasets and _iter_datasets. But let me know how (if) you were using this one.
- get_geography: This is too specific. The ProjectConfig class has `get_base_dimension` which should be used instead.
- get_mapping_type: I think this is stale and completely incorrect. There is no MappingType in our repo.

Reply:
Yeah you are right, the one I used was get_dataset. I even had a TODO in my branch to delete get_geography for the same reason, and yeah, get_mapping_type/get_dimension_mappings is not working.

Reply:
I do use get_project_dimension,  and _iter_base_dim is part of that.

Reply:
Okay then :)